elasticsearch_opentracing/__init__.py

```diff
@@ -64,33 +53,28 @@ def perform_request(self, method, url, params=None, body=None):
         if g_trace_prefix is not None:
             op_name = str(g_trace_prefix) + url
 
-        span = g_tracer.start_span(op_name, child_of=get_active_span())
-        span.set_tag('component', 'elasticsearch-py')
-        span.set_tag('db.type', 'elasticsearch')
-        span.set_tag('span.kind', 'client')
-        span.set_tag('elasticsearch.url', url)
-        span.set_tag('elasticsearch.method', method)
-
-        if body:
-            span.set_tag('db.statement', body)
-        if params:
-            span.set_tag('elasticsearch.params', params)
-
-        try:
-            rv = super(TracingTransport, self).perform_request(method, url, params, body)
+        with g_tracer.start_active_span(op_name) as scope:
+            span = scope.span
+            span.set_tag(tags.COMPONENT, 'elasticsearch-py')
+            span.set_tag(tags.DATABASE_TYPE, 'elasticsearch')
+            span.set_tag(tags.SPAN_KIND, tags.SPAN_KIND_RPC_CLIENT)
+            span.set_tag('elasticsearch.url', url)
+            span.set_tag('elasticsearch.method', method)
+
+            if body:
+                span.set_tag(tags.DATABASE_STATEMENT, body)
+            if params:
+                span.set_tag('elasticsearch.params', params)
+
+            try:
+                rv = super(TracingTransport, self).perform_request(method, url, params, body)
+            except Exception as exc:
+                span.set_tag('error', True)
+                span.set_tag('error.object', exc)
+                raise
```

Comment:
Globally disabling this instrumentation in an error condition seems undesired for a number of reasons.  This is a breaking change though, and if it's determined the original behavior is desired at this point I can revert and open another PR w/ my proposed change.

Reply:
No, this is fine, and I honestly don't think anybody was using this inusual feature.

Reply:
(Also, I will be removing this feature - mostly because I truly think nobody was using it ;) )